docs/tutorials/gpu_executor.md

```diff
@@ -0,0 +1,302 @@
+
+You have probably also noticed that there was a delay (about 3 seconds) when creating the flow.
+This occured because the weights of our model needed to be transfered from CPU to GPU when we
+initialized the executor. However, this action only occurs once in the lifetime of the executor,
+so for most use cases this is not something we would worry about.
+
+## Using GPU in a container
+
+When you'll be using your executor in production you will most likely want to put it in a Docker container, to provide proper environment isolation and to be able to use it easily on any device.
+
+Using GPU-enabled executors in this case is no harder than using them locally. In this case we don't even need to modify the default `Dockerfile`.
+
+```{admonition} Choosing the right base image
+
+In our case we are using the default `jinaai/jina:latest` base image. However, parallel to the comments about having to install CUDA locally, you might need to use a different base image, depending on the framework you are using.
+
+If you need to have CUDA installed in the image, you usually have two options: either you take the `nvidia/cuda` for the base image, or you take the official GPU-enabled image of the framework you are using, for example, `tensorflow/tensorflow:2.6.0-gpu`.
+```
+
+The other file we care about in this case is `config.yml`, and here the default version works as well. So let's build the docker image
+
+```bash
+docker build -t sentence-encoder .
+```
+
+You can run the container to quickly check that everything is working well
+
+```bash
+docker run sentence-encoder
+```
+
+Now, let's use the docker version of our encoder with the GPU. If you've dealt with GPUs in containers before, you probably remember that to use a GPU insite the container you need to pass `--gpus all` command to `docker run`. And jina enables you to do just that.
+
+Here's how we need to modify our `main.py` script to use a GPU-base docker executor
+
+```python
+from jina import Document, DocumentArray, Flow
+
+from executor import SentenceEncoder
+
+def generate_docs():
+    for _ in range(10000):
+        yield Document(
+            text='Using a GPU enables you to significantly speed up encoding'
+        )
+
+f = Flow().add(
+    uses='docker://sentence-encoder', uses_with={'device': 'cuda'}, gpus='all'
+)
+with f:
+    f.post(on='/encode', inputs=generate_docs, show_progress=True, request_size=32)
+```
+
+If we run this with `python main.py`, we'll get the same output as before, except that now we'll also get the output from the Docker container.
+
+You may notice that every time we start the executor, the transformer model gets downloaded again. To speed this up, we would want the encoder to load the model from a file which we have pre-downloaded to our disk.
```

Comment:
wrap in a not admonition ?

Reply:
This connects to the example below as well. I think it's the "main" part of the text, so it's better if it is part of text, rather than an admonition